### PR TITLE
SB-14970 - Draft - Cert-reissue

### PR DESCRIPTION
**Pull request by Hari-stackroute**

### CODE REVIEW

File: actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java [hunk 1]
```diff
@@ -302,6 +302,13 @@ private Map getCertificateDetails(String certificatedId) {
       List responseList = (List) record.get(JsonKey.RESPONSE);
       if (!responseList.isEmpty()) {
```
Comment (manzarul) [inline actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java]: Use CollectionUtils.isnotempty method.

File: actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java [hunk 2]
```diff
@@ -317,32 +324,56 @@ private Map getCertificateDetails(String certificatedId) {
...
   private void addCertificate(Request request) throws JsonProcessingException {
+    Response response = null;
     Map<String, String> storeMap = new HashMap<>();
+    Map<String, Object> telemetryMap = new HashMap();
     Map<String, Object> certAddReqMap = request.getRequest();
     String userId = (String) certAddReqMap.get(JsonKey.USER_ID);
+    String oldCertId = (String)certAddReqMap.get(JsonKey.OLD_ID);
     User user = userService.getUserById(userId);
     assureUniqueCertId((String) certAddReqMap.get(JsonKey.ID));
+    if(null != oldCertId) {
```
Comment (manzarul) [inline actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java]: Use StringUtils.notbalnk

File: actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java [hunk 3]
```diff
@@ -317,32 +324,56 @@ private Map getCertificateDetails(String certificatedId) {
...
+      getCertificateDetails(oldCertId);
+    }
     populateStoreData(storeMap, certAddReqMap);
     certAddReqMap.put(JsonKey.STORE, storeMap);
     certAddReqMap = getRequiredRequest(certAddReqMap);
     certAddReqMap.put(JsonKey.CREATED_AT, getTimeStamp());
-    Response response =
-        cassandraOperation.insertRecord(
-            certDbInfo.getKeySpace(), certDbInfo.getTableName(), certAddReqMap);
+    if(StringUtils.isNotBlank(oldCertId)) {
```
Comment (manzarul) [inline actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java]: above we already have the check , so we can put this code there itself.

File: actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java (hunk 2 above)
Comment (Hari-stackroute) [reply actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java]: done

File: actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java (hunk 1 above)
Comment (Hari-stackroute) [reply actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java]: Done

File: actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java (hunk 3 above)
Comment (Hari-stackroute) [reply actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java]: yes no need second condition again.

**Pull request merged**